### ISSUE TITLE
Cell fill from a lower row overlapping onto top hline

### ISSUE THREAD

**Issue by PgBiel** [PgBiel/typst-tablex#4]
Generating smaller hlines than normal (missing the bottom half of the stroke).

![image](https://user-images.githubusercontent.com/9021226/231345825-cbb1cf95-e07b-4984-8d95-5cd3416ec696.png)

**Comment by PgBiel** [comment]
This might require #2 for a proper solution.

**Comment by Fabioni** [comment]
This really needs to get fixed. Without `fill` is not useable.

**Comment by Fabioni** [comment]
I found a workaround. Having a 0cm row that spans all columns and a 0cm column that spans all rows fixes the issue.

This looks correct:
```
#tablex(
    fill: luma(240),
    columns: (0cm, 1fr, 1fr),
    rowspanx(3)[], [?], [?],
    (), [?], [?], 
    (), [?], [?], 
)
```

but this does not:
```
#tablex(
    fill: luma(240),
    columns: (0cm, 1fr, 1fr),
    (), [?], [?],
    (), [?], [?], 
    (), [?], [?], 
)
```

**Comment by PgBiel** [comment]
Yep, that will work (as will setting `header-rows` to a really high value) because it will force tablex to draw all of your cells at the same time (instead of drawing per row), drawing cells first and then lines, which circumvents this problem. However, that will make it impossible to split your table across pages.

I plan on tackling this on 0.1.0 by providing a new renderer which will draw per page instead of per row, which is how native tables do it, but I haven't gotten to it yet (I also haven't yet experimented with the idea to know how much of it is possible).
